capstone/src/main/webapp/youtubeScript.js

```diff
@@ -34,8 +34,13 @@ const YOUTUBE_CATEGORIES = {
   'Sports' : 17,
 };
 
+/** These variables will keep track of the data required for CSV output */
+let ATTRIBUTE_DATA = [];
```

Comment:
nit: Instead of having these as class constants, it might be more intuitive to pass them in as parameters specific functions. That's just a style preference though

Reply:
+1 to passing these as variables.  Right now having them at global scope works, but if you ever ended up using the `callYouTube` function multiple times (maybe in parallel) you could run into some issues with them as globals.

Also I'd suggest changing the name from ALL_CAPS to just `attributeData`.  I think we want to reserve ALL_CAPS for constants (like BASE_URL in the other files)

Reply:
I understand not having class constants and passing data into functions is superior for scalability and parallel calls. I have revised my implementation in "remove global variable dependence for CSV". There is no longer the need to have variables in global scope for the CSV output to be functional.